compiler/luci/pass/src/FoldFullyConnectedPass.cpp

```diff
@@ -0,0 +1,201 @@
+  RETURN_FALSE_UNLESS(bias != nullptr or no_bias != nullptr);
+
+  RETURN_FALSE_UNLESS(input->dtype() == loco::DataType::FLOAT32);
+  RETURN_FALSE_UNLESS(weights->dtype() == loco::DataType::FLOAT32);
+  if (bias)
+    RETURN_FALSE_UNLESS(bias->dtype() == loco::DataType::FLOAT32);
+
+  auto const input_elems = input->size<loco::DataType::FLOAT32>();
+
+  RETURN_FALSE_UNLESS(weights->rank() == 2);
+  RETURN_FALSE_UNLESS(input_elems % weights->dim(1).value() == 0);
+  auto const batch_size = input_elems / weights->dim(1).value();
+  auto const num_units = weights->dim(0).value();
+
+  if (bias)
+    RETURN_FALSE_UNLESS(bias->size<loco::DataType::FLOAT32>() == num_units);
+
+  tflite::FullyConnectedParams params{};
+  if (!set_kernel_parameters(&params, node))
+    return false; // Unsupported kernel parameter values
+
+  std::vector<uint32_t> output_shape;
+  if (node->keep_num_dims() == false)
+  {
+    output_shape.push_back(batch_size);
+    output_shape.push_back(num_units);
+  }
+  else
+  {
+    output_shape.resize(input->rank());
+    for (uint32_t i = 0; i < input->rank(); i++)
+      output_shape[i] = input->dim(i).value();
+    output_shape[input->rank() - 1] = num_units;
+  }
+
+  auto constant = node->graph()->nodes()->create<luci::CircleConst>();
+  {
+    constant->name(node->name());
+    constant->dtype(node->dtype());
+    constant->rank(node->rank());
+    constant->shape_status(luci::ShapeStatus::VALID);
+    uint32_t num_elem = 1;
+    for (uint32_t i = 0; i < node->rank(); ++i)
+    {
+      constant->dim(i).set(node->dim(i).value());
+      num_elem *= node->dim(i).value();
+    }
+    constant->size<loco::DataType::FLOAT32>(num_elem);
+  }
+
+  auto tensor_shape = [](luci::CircleNode *node) {
+    if (node == nullptr)
+      return tflite::RuntimeShape();
+
+    tflite::RuntimeShape runtime_shape(node->rank());
+    for (uint32_t i = 0; i < node->rank(); ++i)
+      runtime_shape.SetDim(i, node->dim(i).value());
+    return runtime_shape;
+  };
+
+  auto tensor_data = [](luci::CircleConst *node) -> float * {
+    if (node == nullptr)
+      return nullptr;
+
+    return &node->at<loco::DataType::FLOAT32>(0);
+  };
+
+  tflite::reference_ops::FullyConnected(
+    params, tensor_shape(input), tensor_data(input), tensor_shape(weights), tensor_data(weights),
+    tensor_shape(bias), tensor_data(bias), tensor_shape(constant), tensor_data(constant));
+
+  loco::replace(node).with(constant);
+
+  return true;
+}
+
+} // namespace
+
+namespace luci
+{
+
+/**
+ * Constant Folding for FullyConnected Op
+ **/
+bool FoldFullyConnectedPass::run(loco::Graph *g)
+{
+  bool changed = false;
+  for (auto node : loco::active_nodes(loco::output_nodes(g)))
+  {
+    auto fc = dynamic_cast<CircleFullyConnected *>(node);
+
+    if (fc == nullptr)
+      continue;
+
+    if (fc->dtype() != loco::DataType::FLOAT32)
+      continue;
+
+    changed = fold_fully_connected(fc);
```

Comment:
Is it okay with just `=`?
```suggestion
    changed |= fold_fully_connected(fc);
```

Reply:
Thanks for finding!